lib/core/src/Cardano/Wallet/DB/Store/Wallets/Model.hs

```diff
@@ -90,3 +126,79 @@ linkedTransactions (TxMetaHistory m) = Map.keysSet m
 
 walletsLinkedTransactions :: Map W.WalletId TxMetaHistory -> Set TxId
 walletsLinkedTransactions = Set.unions . toList .  fmap linkedTransactions
+
+mkTransactionInfo :: Monad m
+    => TimeInterpreter m
+    -> W.BlockHeader
+    -> TxRelationF 'With
+    -> DB.TxMeta
+    -> m WT.TransactionInfo
+mkTransactionInfo ti tip TxRelationF {..} DB.TxMeta {..} = do
+    txTime <- interpretQuery ti . slotToUTCTime $ txMetaSlot
+    return
+        $ WT.TransactionInfo
+        { WT.txInfoId = getTxId txMetaTxId
+        , WT.txInfoFee = WC.Coin . fromIntegral <$> txMetaFee
+        , WT.txInfoInputs = mkTxIn <$> ins
+        , WT.txInfoCollateralInputs = map mkTxCollateral collateralIns
+        , WT.txInfoOutputs = mkTxOut <$> outs
+        , WT.txInfoCollateralOutput = fmap mkTxCollateralOut collateralOuts
```

Comment:
```suggestion
        , WT.txInfoCollateralOutput = mkTxCollateralOut <$> collateralOuts
```

Reply:
Excellent! 🥳 The `Map TxId (TxRelationF f)` model simplified a few things:

* the `filter ((txid ==) …)` stuff is gone
* the `Maybe` type for the collateral output eliminates the comment about correctness.

Reply:
Yes it worked out in the end :-)